cpu/efm32/periph/adc_series2.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2022 SSV Software Systems GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_efm32
+ * @ingroup     drivers_periph_adc
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Juergen Fitschen <me@jue.yt>
+ *
+ * @}
+ */
+
+#include <assert.h>
+
+#include "cpu.h"
+#include "mutex.h"
+
+#include "periph_conf.h"
+#include "periph/adc.h"
+#include "periph/gpio.h"
+
+#include "em_cmu.h"
+#include "em_gpio.h"
+#include "em_iadc.h"
+
+static mutex_t adc_lock[ADC_DEV_NUMOF];
+
+int adc_init(adc_t line)
+{
+    /* check if line is valid */
+    if (line >= ADC_NUMOF) {
```

Comment:
```suggestion
    assert(line < ADC_NUMOF);
```
should also do here

Reply:
df60c545ee454a77688e2239b32060167aa3aa60